AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1700,6 +1702,10 @@ - (void)__setSupernode:(ASDisplayNode *)newSupernode
       [self exitHierarchyState:stateToEnterOrExit];
     }
   }
+    
+  if ([newSupernode supportsUpwardPropagation]) {
```

Comment:
This looks great. We may not need to revisit this API, the new name is pretty strong for long-term use unless we encounter a new case that is unusual to implement with it